core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,270 @@
+import org.kiwix.sharedFunctions.bookOnDiskEntity
+
+internal class NewBookmarksDaoTest {
+
+  private val box: Box<BookmarkEntity> = mockk(relaxed = true)
+  private val newBookmarksDao = NewBookmarksDao(box)
+
+  @BeforeEach
+  fun init() {
+    clearAllMocks()
+    mockkStatic(CoreApp::class)
+    every { CoreApp.getInstance().packageName } returns "pkg"
+  }
+
+  @Test
+  fun `get bookmarks without current book`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(false, zimFileReader))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `get bookmarks from current book when zimFileReader is null`() {
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimName, "") } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(true, null))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `get bookmarks from current book`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimName = "zimName"
+    every { zimFileReader?.name } returns zimName
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader?.name) } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(false, zimFileReader))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns list of bookmarkUrl`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimId = ""
```

Comment:
these values are indistinct from whats get returned if ZimFileReader is null making the test less legible